cmd/root.go

```diff
@@ -253,3 +278,82 @@ func makeRepo(uri string) (clients.Repo, error) {
 
 	return nil, fmt.Errorf("unable to parse as github, gitlab, or azuredevops: %w", compositeErr)
 }
+
+// processRepo performs the scanning and formatting for a single repo URI.
+// It returns the Result when successful (or when combined output is requested),
+// or an error describing why the URI should be skipped.
+func processRepo(
+	ctx context.Context,
+	uri string,
+	o *options.Options,
+	enabledProbes []string,
+	enabledChecks checker.CheckNameToFnMap,
+	opts []scorecard.Option,
+	checkDocs docs.Doc,
+	pol *policy.ScorecardPolicy,
+) (*scorecard.Result, error) {
+	var repo clients.Repo
+	var err error
+
+	if o.Local != "" && uri == o.Local {
+		repo, err = localdir.MakeLocalDirRepo(uri)
+		if err != nil {
+			return nil, fmt.Errorf("localdir: %w", err)
+		}
+	} else {
+		repo, err = makeRepo(uri)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Start banners with repo uri (show banners in default format only or combined mode)
+	if o.Format == options.FormatDefault || o.Format == options.FormatCombined {
+		if len(enabledProbes) > 0 {
+			printProbeStart(uri, enabledProbes)
+		} else {
+			printCheckStart(uri, enabledChecks)
+		}
+	}
+
+	result, err := scorecard.Run(ctx, repo, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("run: %w", err)
+	}
+
+	result.Metadata = append(result.Metadata, o.Metadata...)
+
+	// Stable order
+	sort.Slice(result.Checks, func(i, j int) bool {
+		return result.Checks[i].Name < result.Checks[j].Name
+	})
+
+	// End banners BEFORE RESULTS (always show banners even in combined-only mode)
+	if o.Format == options.FormatDefault {
+		if len(enabledProbes) > 0 {
+			printProbeResults(uri, enabledProbes)
+		} else {
+			printCheckResults(uri, enabledChecks)
```

Comment:
as written, this only prints the results for the default format, but the comment makes me think it should also apply to the combined format:

```
if o.Format == options.FormatDefault || o.Format == options.FormatCombined {
```